added support for codecarbon integration, integration examples, html …

jzvolensky opened the pull request:
Codecarbon testing done some time ago. Contains 3 example energy use comparisons.

clausmichele commented:
Easy things to fix:
1. remove the emissions.csv file in the root folder
2. rename the utils.py file to something more unique (codecarbon_utils.py ?) and modify the notebook call accordingly

Other things:
- We don't need to show two ways to use the library, keep one or the other (decorator)
- Keep the simple example using the matrix multiplication to introduce the library
- Introduce another example using openEO. You can reuse the NDVI example at the bottom here: https://github.com/jzvolensky/cubes-and-clouds.git or modify it to compute the NDSI (which fits better the other examples), see this notebook where ndsi is computed: https://github.com/EO-College/cubes-and-clouds/blob/main/lectures/3.1_data_processing/3.1_exercises/_alternatives/31_data_processing_stac.ipynb
- The topic should also be covered in the lecture: please add some content in https://github.com/EO-College/cubes-and-clouds/blob/main/lectures/2.4_formats_and_performance/2.4_formats_and_performance.md about this (why it's important to know the energy consumption? -> users should be aware of their energy/carbon footprint also when using cloud services, ...)
- Think about a question or two we can add to the quiz (at the bottom of lecture markdown file).

clausmichele commented on lectures/2.4_formats_and_performance/exercises/24_formats_and_performance.ipynb:
```diff
@@ -0,0 +1,379 @@
+{
```
<p>Remove repetition: <em>In this example we need to squeeze -&gt; We need to squeeze</em></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/EO-College/cubes-and-clouds/pull/77/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

clausmichele replied:
<p>Remove the first spatial_extent, not used and overwritten in the subsequent line</p><p>Remove s2_cube.execute() (unused)</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/EO-College/cubes-and-clouds/pull/77/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

jzvolensky commented:
I removed the CPU TDP option because in this latest setup we are using it doesnt actually do anything so we default to 40 or something as per the codecarbon docs

I also added a check to remove a `.lock` file because sometimes the codecarbon process does not exit fully and then it breaks. This happens when you init, restart the kernel, init again. This way it should always be just 1 instance.

clausmichele commented:
A last feedback:

- Provide a sub-title to the exercise under the lecture title
![immagine](https://github.com/user-attachments/assets/8352fb28-6fbd-4d71-b172-411cfdb02a8a)
- Think about 1-2 questions to be added in the quiz, which refers to the exercise. (If the user need to remember values provided in the exercise, mark them with a bold text as I did in the exercises of the previous lecture)

After the points are being addressed, we can finally merge this.
